Added Issue and Pull Request templates

GrooveCS opened the pull request:
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

Add Issue and Pull Request Template
<!-- Describe your changes here- ideally, you can get that description straight from your descriptive commit message(s)! -->

## Issue reference

We strive to have all PR being opened based on an issue where the problem or feature has been discussed before implementation.

Please reference the issue this PR will close: #_[issue number]_

## Submitter Checklist

**Documentation PR**
Code Review not required.

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Code compiles correctly
- [ ] Created/updated tests
- [ ] Extended the documentation
- [ ] Includes [tests](https://github.com/~) (if functionality changed/added)
- [ ] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [ ] Commit messages follow [commit message best practices](https://github.com/~)

_See [the contribution guide](https://github.com/~/CONTRIBUTING.md) for more details._

## Release Notes

```release-note
NONE
```
<!--
Does your PR contain User facing changes?

If so, briefly describe them here so we can include this description in the
release notes for the next release!

For pull requests with a release note:

```release-note
Your release note here
```

For pull requests that require additional action from users switching to the new release, include the string "action required" (case insensitive) in the release note:

```release-note
action required: your release note here
```

For pull requests that don't need to be mentioned at release time, use the `/release-note-none` Prow command to add the `release-note-none` label to the PR. You can also write the string "NONE" as a release note in your PR description:

```release-note
NONE
```
-->

GrooveCS commented:
@RinkiyaKeDad 

If this is a sub-project, please update the URL hyperlinks to the parent test, commit, and contributor guides.

RinkiyaKeDad commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,50 @@
+
+## Version & Operating System
+
+- Dependency Analyzer:
```
The repository and tool name might change soon. We'll probably be calling the tool `depstat` 😄

GrooveCS replied:
Sounds good.

RinkiyaKeDad commented:
Thanks for the PR @GrooveCS! 
IMO the templates seem a bit too complicated which might scare off new contributors, could we go with something simple instead? 😅 That said, I think @nikhita would be the best person to review this :)

GrooveCS commented:
Hi @RinkiyaKeDad & @Nikhita

Please review the updated documents, and provide feedback, suggestions and concerns. 

- PR [WIP] [Added Issue and Pull Request templates](https://github.com/kubernetes-sigs/depstat/pull/15) #15

Thanks @GrooveCS

RinkiyaKeDad commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,30 @@
+
+**Does this issue reproduce with the latest release?**
+
+**Which depstat CLI options are you using?**
```
I think we can remove this since this will be answered in "What did you do?" anyways.

RinkiyaKeDad commented on .github/ISSUE_TEMPLATE/discussion.md:
```diff
@@ -0,0 +1,8 @@
+---
+name: Feature Request
+about: Start a discussion for depstat dependency analyzer CLI
```
We can now use just "depstat" instead of "depstat dependency analyzer CLI" :)

RinkiyaKeDad commented on .github/ISSUE_TEMPLATE/proposal.md:
```diff
@@ -0,0 +1,20 @@
+---
+name: Proposal
+about: Create a proposal for depstat dependency analyzer CLI
```
Same here.

RinkiyaKeDad commented on cmd/completion.go:
```diff
@@ -0,0 +1,87 @@
+/*
```
I think you accidentally committed changes from your other PR here :)

RinkiyaKeDad commented on go.mod:
```diff
@@ -2,4 +2,4 @@ module github.com/kubernetes-sigs/depstat
 go 1.16
 
-require github.com/spf13/cobra v1.1.1
+require github.com/spf13/cobra v1.1.3
```
Refer to [this](https://github.com/kubernetes-sigs/depstat/pull/20#discussion_r621839266).

RinkiyaKeDad reviewed (changes requested):
Thanks! Looks all good to me except a few things for which I've left some notes. Once you make the changes we can request a final review from Nikhita 🙂

RinkiyaKeDad commented on cmd/completion.go:
```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
Hi @GrooveCS! Can you please remove this file from the PR before we can merge this?

GrooveCS replied:
Understood. I it will be done shortly.

GrooveCS committed:
Added Issue and Pull Request templates

**Issue Templates**
1. `github/ISSUE_TEMPLATE/bug_report.md`
1. `.github/ISSUE_TEMPLATE/discussion.md`
1. `.github/ISSUE_TEMPLATE/feature_request.md`
1. `.github/ISSUE_TEMPLATE/proposal.md`
1. `.github/ISSUE_TEMPLATE/question.md`

**Pull Request Template**
- `.github/pull_request_template.md`
Added Issue and Pull Request templates

**Issue Templates**
1. `github/ISSUE_TEMPLATE/bug_report.md`
1. `.github/ISSUE_TEMPLATE/discussion.md`
1. `.github/ISSUE_TEMPLATE/feature_request.md`
1. `.github/ISSUE_TEMPLATE/proposal.md`
1. `.github/ISSUE_TEMPLATE/question.md`

**Pull Request Template**
- `.github/pull_request_template.md`

RinkiyaKeDad reviewed (commented):
This looks good to me now. Please revert it back to a normal PR instead of a draft (and remove the "WIP"). Will request a final review from @nikhita before merging. Thanks!

nikhita commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,28 @@
+
+depstat version:
+
+**Does this issue reproduce with the latest release?**
```
Is there a particular reason we are asking this explicitly?

If the author of the bug report would have tested against the latest depstat release, that would be captured in the first question, correct? :thinking:

GrooveCS replied:
Aggreed. I based the template on other in the Kubernetes project world, and will remove line 16.

nikhita commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,28 @@
+---
+name: Bug report
+about: Create a bug report to help improve depstat
```
```suggestion
about: Problems and issues with code or docs
```

nikhita commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,28 @@
+---
```
For this one, I think we can use the kubernetes bug report template directly for consistency - https://github.com/kubernetes/kubernetes/blob/master/.github/ISSUE_TEMPLATE/bug-report.md

GrooveCS replied:
Sounds good. I will swap it out shortly.

nikhita commented on .github/ISSUE_TEMPLATE/discussion.md:
```diff
@@ -0,0 +1,8 @@
+name: Feature Request
+about: Start a discussion for depstat
+title: ''
+labels: kind/discussion
```
Please don't use issues for discussions. Discussions should always go to slack, mailing list or discuss.k8s.io

GrooveCS replied:
Didn't know, and now I do. Thanks for the insight. It was added out of habit, and learning what is the norm in other related projects.

nikhita commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,26 @@
+**Describe the solution you'd like**
+<!-- A clear and concise description of what you want to happen. -->
+
+**Describe alternatives you've considered**
```
Love that this is explicitly called out! :100:

nikhita commented on .github/ISSUE_TEMPLATE/proposal.md:
```diff
@@ -0,0 +1,20 @@
+name: Proposal
+about: Create a proposal for depstat
+title: ''
+labels: kind/proposal
```
`kind/feature` and `kind/proposal` are the same thing, right?

None of the k8s repos use `kind/proposal` so I'd suggest removing this.

nikhita commented on .github/ISSUE_TEMPLATE/question.md:
```diff
@@ -0,0 +1,9 @@
+name: Question
+about: Ask a question about depstat
+title: ''
+labels: kind/question
```
Again, please don't use GitHub issues for questions or discussions. Support questions should go to discuss.kubernetes.io.

You can set up a config.yml for this - https://github.com/kubernetes/kubernetes/blob/master/.github/ISSUE_TEMPLATE/config.yml

nikhita commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,52 @@
+<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->
+
+# Changes
```
I'd recommend using Kubernetes' pull request template - https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md

The k8s template is used in a lot of other repos. Would be good to use it for consistency.

nikhita reviewed (commented):
@GrooveCS Left some comments, but looks good overall and thanks for your patience in getting this merged :raised_hands:

RinkiyaKeDad commented:
tide/merge-method-squash

RinkiyaKeDad commented:
/tide/merge-method-squash

GrooveCS committed:
Update .github/ISSUE_TEMPLATE/bug_report.md

Modify the about to be `Problems and issues with code or docs`

Co-authored-by: Nikhita Raghunath <nikitaraghunath@gmail.com>
